File: app/core/Player.scala
package core

import Race._
import models.format.{AiType, PlayerType}
import AiType._
import models.format.PlayerType

class Player(val p: PlayerType.Value, val number: Int, val race: Race, val aiType: AiType, val startResources: (Int, Int, Int),
              val startPos: (Int, Int)) {
  // todo upgrades

}

File: public/javascripts/player.js
function Player(type, race, aiType, startRes, color) {
    this.type = PlayerType[type];
    switch (race) {
        case 0: this.race = Race.HUMAN; break;
        case 1: this.race = Race.ORC; break;
        case 2: this.race = "Neutral";
    }

    this.aiType = aiType;
    this.res = startRes;
    this.color = color;
}